cpp/src/arrow/array/builder_dict.h

```diff
@@ -724,6 +724,7 @@ using BinaryDictionaryBuilder = DictionaryBuilder<BinaryType>;
 using StringDictionaryBuilder = DictionaryBuilder<StringType>;
 using BinaryDictionary32Builder = Dictionary32Builder<BinaryType>;
 using StringDictionary32Builder = Dictionary32Builder<StringType>;
+using LargeBinaryDictionary32Builder = Dictionary32Builder<LargeBinaryType>;
```

Comment:
Does `LargeStringDictionary32Builder` worth being added as well?

Reply:
Maybe we leave it for a follow up PR or implement it when needed?

Reply:
Why not add it here? It should be relatively trivial.

Reply:
added